src/commands/quit.cpp

```diff
@@ -6,12 +6,21 @@
 /*   By: mbernard <mbernard@student.42lyon.fr>      +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/10/30 10:10:53 by mbernard          #+#    #+#             */
-/*   Updated: 2024/10/30 10:11:41 by mbernard         ###   ########.fr       */
+/*   Updated: 2024/11/04 09:30:37 by mbernard         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "../includes/Parser.hpp"
 
-void Parser::quit(Client& client, std::string arg) {
-  
+void Server::quit(std::string argument, Client &client, clientsMap &cltMap) {
+    if (argument.empty()) {
+        clearClient(client.getFd());
+        return;
+    }
+    std::string message = argument.substr(argument.find_first_of(":"));
+    clearClient(client.getFd());
+    // cltMap.erase(client.getFd());
+    for (clientsMap::iterator it = cltMap.begin(); it != cltMap.end(); ++it) {
+        it->second.receiveMessage("QUIT :" + message + "\r\n");
```

Comment:
Message should be sent with the client information who has quit (`nickname!username@hostname`)

e.g.
```
:dan-!d@localhost QUIT :Quit: Bye for now!
# dan- is exiting the network with the message: "Quit: Bye for now!"
```